fieldpath/set.go

```diff
@@ -503,3 +638,50 @@ func (s *SetNodeMap) Leaves() *SetNodeMap {
 	}
 	return out
 }
+
+// Filter defines an interface for filtering a set.
```

Comment:
It's a little confusing to someone not steeped in this area - some of these are talking about EXCLUDE and some are talking about INCLUDE - is there a way to make them clearer?  Maybe I am just not the audience for it :)

Reply:
I've expanded this comment out significantly. Hopefully that makes it more clear.